tests/PySys/docker_plugin/docker_plugin_install_with_version/run.py

```diff
@@ -0,0 +1,38 @@
+import sys
+
+from environment_docker_plugin import DockerPlugin
+
+"""
+Validate docker plugin install
+
+Using `registry`[https://hub.docker.com/_/registry] as a test container
+
+"""
+
+
+class DockerPluginInstallWithVersionTest(DockerPlugin):
+
+    image_name = "registry"
+    image_version = "2.7.1"
+
+    def setup(self):
+        super().setup()
+        # Assert that an image with the given name is not present on the machine before test
+        self.assert_image_present(
+            self.image_name, negate=True, abortOnError=True)
+        # Assert that no containers using the given image name and version are running
+        self.assert_container_running(
+            self.image_name, negate=True, abortOnError=True)
+
+        # Register routines to cleanup containers and images added during test
+        self.addCleanupFunction(self.cleanup_containers)
+        self.addCleanupFunction(self.cleanup_images)
```

Comment:
Since all the tests that derives `DockerPlugin` also includes these lines, I wonder if these tests can inherit these clean-up routines. Would be nicer!